fathom/src/surface/grammar.lalrpop

```diff
@@ -132,22 +134,18 @@ FunTerm: Term<'arena, ByteRange> = {
             scope.to_scope(body_type),
         )
     },
-    <start: @L> "fun" <param_pattern: AnnPattern> "->"  <body_type: FunTerm> <end: @R> => {
-        let (param_pattern, param_type) = param_pattern;
+    <start: @L> "fun" <patterns: AnnPattern+> "->"  <output_type: FunTerm> <end: @R> => {
```

Comment:
I just realised functions have one or more arguments. Is that because zero argument function are constants, or something else?

Reply:
Yeah, the multiple argument syntax is sugar for nested, single argument functions, so a function with no arguments would be superfluous. Perhaps it could be added for completeness at some stage… I dunno 😅